src/Microsoft.ServiceFabric.Services/Communication/Client/CommunicationClientFactoryBase.cs

```diff
@@ -512,6 +508,12 @@ private static bool IsValidRsp(CommunicationClientCacheEntry<TCommunicationClien
                         cacheEntry.Semaphore.Release();
                     }
 
+                    if (retry)
+                    {
+                        await Task.Delay(retryDelay, cancellationToken);
+                        continue;

```

Comment:
Shouldnt you be setting retry to false here?(otherwise, wouldnt this become an infinite loop?)

Reply:
Nvm. Retry is scoped to the while.

---
In reply to: [250832599](https://github.com/Microsoft/service-fabric-services-and-actors-dotnet/pull/170#discussion_r250832599) [](ancestors = 250832599)